libbeat/processors/add_kubernetes_metadata/kubernetes.go

```diff
@@ -91,11 +104,30 @@ func New(cfg *common.Config) (processors.Processor, error) {
 		return nil, fmt.Errorf("Can not initialize kubernetes plugin with zero matcher plugins")
 	}
 
+	processor := &kubernetesAnnotator{
+		indexers: indexers,
+		matchers: matchers,
+		cache:    newCache(config.CleanupTimeout),
+	}
+
 	client, err := kubernetes.GetKubernetesClient(config.KubeConfig)
 	if err != nil {
-		return nil, err
+		if config.InCluster {
+			logp.Err("%v: could not create kubernetes client using in_cluster config", "add_kubernetes_metadata")
+		} else {
+			logp.Err("%v: could not create kubernetes client using config: %v", "add_kubernetes_metadata", config.KubeConfig)
+		}
+		processor.kubernetesAvailable = false
```

Comment:
Nit. Set `kubernetesAvailable` to `false` already when initializing the `processor` (or just don't set it as is the zero value), and only set it to true when all checks have been done.

Reply:
Also here it is not needed to set `processor.kubernetesAvailable = false`.
```suggestion
```